File: onebot-sdk/src/main/java/cn/evole/onebot/sdk/event/IgnoreEvent.java
package cn.evole.onebot.sdk.event;

import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.experimental.SuperBuilder;

/**
 * Description:
 * Author: cnlimiter
 * Date: 2022/10/3 13:20
 * Version: 1.0
 */
@Data
@SuperBuilder(toBuilder = true)
@EqualsAndHashCode(callSuper = true)
public class IgnoreEvent extends Event{
    public IgnoreEvent(){
        this.setPostType("IGNORED");
        this.setTime(System.currentTimeMillis());
    }
    public IgnoreEvent(long id){
        this.setPostType("IGNORED");
        this.setTime(System.currentTimeMillis());
        this.setSelfId(id);
    }


}

File: onebot/src/main/kotlin/sdk/event/IgnoreEvent.kt
package cn.evolvefield.onebot.sdk.event

import cn.evolvefield.onebot.sdk.event.Event
import lombok.Data
import lombok.EqualsAndHashCode
import lombok.experimental.SuperBuilder

@Data
@SuperBuilder(toBuilder = true)
@EqualsAndHashCode(callSuper = true)
class IgnoreEvent constructor() : Event() {
    init {
        postType = "IGNORED"
        time = System.currentTimeMillis()
    }
    constructor(id: Long) : this() {
        selfId = id
    }
}